Lib/_pyio.py

```diff
@@ -937,10 +937,8 @@ def write(self, b):
             return 0
         pos = self._pos
         if pos > len(self._buffer):
-            # Inserts null bytes between the current end of the file
-            # and the new write position.
-            padding = b'\x00' * (pos - len(self._buffer))
-            self._buffer += padding
+            # Expand the buffer to be able to hold the new bytes.
+            self._buffer.resize(pos + n)
```

Comment:
I don't think that written `n` null bytes is useful, since `self._buffer[pos:pos + n] = b` will extend the buffer anyway to write `b` bytes. I suggest:

```suggestion
            self._buffer.resize(pos)
```

Reply:
I don't have a strong leaning either way. `pos + n`, there's one resize and extra/unnecessary memset. Doing `pos` there will be 2 resizes and no unnecessary memset.

Reply:
> I don't have a strong leaning either way

In this case, I would suggest to avoid the extra `memset(0)` to leave the code as it is.

Maybe a benchmark would help to decide which variant is the most efficient. I suppose that it depends on the buffer size.

Reply:
I'll work on a little benchmark. Either is definitely faster than current  (make a padding bytes and copy) as the `bytearray.resize()` benchmark showed.